src/lib/mod.rs

```diff
@@ -77,59 +79,73 @@ impl HacksawResult {
     }
 }
 
-pub fn set_shape(conn: &xcb::Connection, window: xcb::Window, rects: &[xcb::Rectangle]) {
+pub fn set_shape<C: Connection>(conn: &C, window: xproto::Window, rects: &[xproto::Rectangle]) {
     shape::rectangles(
-        &conn,
-        shape::SO_SET as u8,
-        shape::SK_BOUNDING as u8,
-        0,
+        conn,
+        shape::SO::Set,
+        shape::SK::Bounding,
+        xproto::ClipOrdering::Unsorted,
```

Comment:
What's this about?

Reply:
The function called for a clip ordering enum. https://docs.rs/x11rb/0.5.0/x11rb/protocol/shape/fn.rectangles.html

And 0 is unsorted https://docs.rs/x11rb/0.5.0/src/x11rb/protocol/xproto.rs.html#12963-12968

Reply:
it would be nice to have some explanation of this opaque value, but i understand that this is not a new problem in this pr